hazelcast/src/main/java/com/hazelcast/sql/impl/type/converter/AbstractStringConverter.java

```diff
@@ -129,7 +131,7 @@ public final LocalDate asDate(Object val) {
         try {
             return LocalDate.parse(cast(val));
         } catch (DateTimeParseException e) {
-            throw cannotParseError(QueryDataTypeFamily.DATE);
+            return parseDateWithPossibleLeadingZeroes(val);
```

Comment:
We should use the modified `DateTimeFormatter.ISO_LOCAL_DATE` directly, not after catching an erro. Just change the width from 2 to 1. 

We also need to do this for other types, e.g. time for example allows `12:1` instead of `12:01`.

Reply:
I will introduce separate PR for time, WDYT?

Reply:
I don't mind. But using `DateTimeFormatter` should be done here.